CMakeLists.txt

```diff
@@ -110,20 +110,7 @@ MESSAGE( STATUS "CMAKE_BUILD_TYPE is ${CMAKE_BUILD_TYPE}")
 SET (LIBRARY_OUTPUT_PATH ${qmcpack_BINARY_DIR}/lib CACHE PATH "Single output directory for building all libraries.")
 SET (EXECUTABLE_OUTPUT_PATH ${qmcpack_BINARY_DIR}/bin CACHE PATH "Single output directory for building all executables.")
 
-######################################################################
-# build include and library paths required for the projects
-# For XYZ library, set an environment as
-# export XYZ_HOME='root-of-the-library'
-# XYZ_HOME/include
-# XYZ_HOME/lib
-# IF XYZ library is found, XYZ_FOUND is set to true
-#
-# find_package is used with cmake 2.8 and higher
-######################################################################
-#if(APPLE)
-#  set(QMC_INCLUDE_PATHS ${QMC_INCLUDE_PATHS} /sw/include)
-#  set(QMC_LIBRARY_PATHS ${QMC_LIBRARY_PATHS} /sw/lib)
-#endif(APPLE)
+SET(QMC_CXX_STANDARD 17 CACHE STRING "QMCPACK C++ language standard")
```

Comment:
Unless you have a specific use for this, please no more variables.

Reply:
I really prefer this an option rather than having it hard-coded.
I still need 14 for IBM XL for testing purpose.
Also experiment 20 when the time comes.
Editing CMakeLists.txt is too much interruption.
I can add a warning saying this there is no support if the value is not 17.